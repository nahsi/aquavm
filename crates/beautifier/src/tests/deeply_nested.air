(xor
  (seq
    (seq
       (call "me" ("" "") [] var)
       (ap var.$.value data))
    (par
      (par
        (call "other1" ("set" "") [data])
        (call "other2" ("insert" "") [data]))
      (seq
        (ap "key" key)
        (seq
          (call "other3" ("" "") [key data] $values)
          (fold $values i
            (seq
              (xor
                (seq
                  (call "other2" ("insert" "") [i] j)
                  (call "other3" ("report" "") [j]))
                (fail 3 "fail to fold"))
              (next i)))))))
  (par
    (seq
      (call "other1" ("report" "error") [%last_error%] rep)
      (call "other2" ("report" "error") [%last_error% rep]))
    (call "other3" ("report" "error") [%last_error%])))
