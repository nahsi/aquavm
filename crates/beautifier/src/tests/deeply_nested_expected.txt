try:
    var <- call "me" ("", "") []
    ap var.$.value data
    par:
        par:
            call "other1" ("set", "") [data]
        |
            call "other2" ("insert", "") [data]
    |
        ap "key" key
        $values <- call "other3" ("", "") [key, data]
        fold $values i:
            try:
                j <- call "other2" ("insert", "") [i]
                call "other3" ("report", "") [j]
            catch:
                fail 3 "fail to fold"
            next i
catch:
    par:
        rep <- call "other1" ("report", "error") [%last_error%]
        call "other2" ("report", "error") [%last_error%, rep]
    |
        call "other3" ("report", "error") [%last_error%]
